java-ast/addon/src/main/java/org/jboss/windup/ast/java/ReferenceResolvingVisitor.java

```diff
@@ -455,48 +594,102 @@ public boolean visit(FieldDeclaration node)
                 {
                     ResolveClassnameResult result = resolveClassname(qualifiedName.getFullyQualifiedName().toString());
                     ResolutionStatus status = result.found ? ResolutionStatus.RECOVERED : ResolutionStatus.UNRESOLVED;
-
                     PackageAndClassName packageAndClassName = PackageAndClassName.parseFromQualifiedName(result.result);
 
+                    if (packageAndClassName.constantName != null && packageAndClassName.packageName == null)
+                    {
+                        // the Enum Class and Constant doesn't have to be FQCN as package could miss
+                        packageAndClassName = PackageAndClassName.parseFromQualifiedName(state.getClassNameToFQCN().get(packageAndClassName.className) + "."+packageAndClassName.constantName);
+
+                        //additionally add Enum Constant type reference if found in String
+                        ClassReference reference = new ClassReference(packageAndClassName.toString(),
```

Comment:
I originally was thinking about having a type.isEnum() test, but I reworked the impl and forgot to test it for that? Good catch!